### DISCUSSION TITLE
L10S Ultra Support

### DISCUSSION THREAD

**Discussion by FrancisMurtagh-arm** [Hypfer/Valetudo#1690] [Q&A / Support] [answered]
Hi, 

I've just become aware of this project from this video: https://www.youtube.com/watch?v=r_04K5SPEXI

This is after buying a dreamebot L10s Ultra, do you foresee supporting this model?

I don't see it listed here https://valetudo.cloud/pages/general/supported-robots.html or here https://github.com/Hypfer/Valetudo/blob/master/docs/_pages/general/supported-robots.md

Thanks,
Francis.

**Comment by Hypfer** [comment, marked as answer]
Well I guess now is a good time to answer this question one time and then just point to the answer until there are any news on that topic.
This reflects the situation at 2023-01-17. If anyone reads this, make sure to keep that in mind before you go out in the wild world of online comment sections and tell people what Valetudo does or doesn't support. Thank you.

Dennis and I - well mostly Dennis as far as rooting goes but I'm trying learn from him and support the process :) - are still following the market and occasionally buy robots to look into. This is funded mostly out of our own pockets but can be supported by donating to Dennis: https://builder.dontvacuum.me/donations.txt

In fact, there has been some success so far, however just because Dennis managed to get root doesn't mean that anyone could.
It's a proof-of-concept you could say. A public root for a specific robot is something else entirely to that that takes much more research, work, documentation, tooling and more to be viable, reproducible, safe, etc.

Therefore, there are a few robots in my possession that run Valetudo but aren't listed on the supported robots page because there is no public root available for them and also won't be anytime soon.

This list includes:
- Dreame L10S Ultra
- Dreame L10S Pro
- Xiaomi X10 Plus
- Dreame D10S Pro
- Dreame D10S Plus


Unfortunately, it is not possible to help at this point in time. It is also not possible to test anything. Please refrain from arguing about this.
As an owner of one of these robots, the only thing you can do right now is wait patiently until further notice.

It will take at least a few months so if you don't already have one of those, I'd advise against buying a robot from that list and putting it in storage until then unless you're very _very_ patient. It would likely still be a bad idea though given that prices will likely fall as time goes by.

Fortunately though, as soon as a public root might become available, you should likely be able to use Valetudo just fine, as until then, there will have been quite some time to test things on these robots.
